core/tests/ts-integration/src/context-owner.ts

```diff
@@ -541,17 +541,70 @@ export class TestContextOwner {
         this.reporter.finishAction();
     }
 
+    private async waitForVmPlayground() {
```

Comment:
May I ask you to add somewhat verbose comments on what do these functions do and why do we need them? Because there is a good chance that in the future someone will hard time understanding what's going on in here.